frontend/app/routes/_public+/intake+/$id+/demographics-part2.tsx

```diff
@@ -0,0 +1,63 @@
+import { json } from '@remix-run/node';
+import type { ActionFunctionArgs, LoaderFunctionArgs } from '@remix-run/node';
+import { Form, useLoaderData } from '@remix-run/react';
+
+import { useTranslation } from 'react-i18next';
+
+import { Button, ButtonLink } from '~/components/buttons';
+import { InputRadios } from '~/components/input-radios';
+import { getLookupService } from '~/services/lookup-service.server';
+import { getNameByLanguage, getTypedI18nNamespaces } from '~/utils/locale-utils';
+
+const i18nNamespaces = getTypedI18nNamespaces('demographics-oral-health-questions', 'gcweb');
+
+export const handle = {
+  i18nNamespaces,
+  pageIdentifier: 'CDCP-1112',
+  pageTitleI18nKey: 'demographics-oral-health-questions:part2.page-title',
+};
+
+export async function loader({ request }: LoaderFunctionArgs) {
+  const sexAtBirthTypes = await getLookupService().getAllSexAtBirthTypes();
+
+  return json({ sexAtBirthTypes });
+}
+
+export async function action({ request }: ActionFunctionArgs) {
+  return json('');
+}
+
+export default function DemographicsPart2() {
+  const { sexAtBirthTypes } = useLoaderData<typeof loader>();
+  const { i18n, t } = useTranslation(i18nNamespaces);
+
+  return (
+    <>
+      <Form method="post">
+        <div className="my-6">
+          {sexAtBirthTypes.length > 0 && (
+            <InputRadios
+              id="born-type"
+              name="bornType"
+              legend={t('demographics-oral-health-questions:part2.question6')}
+              options={sexAtBirthTypes.map((sexAtBirthType) => ({
+                children: getNameByLanguage(i18n.language, sexAtBirthType),
+                value: sexAtBirthType.id,
+              }))}
+              required
+            />
+          )}
+        </div>
+
+        <div className="flex flex-wrap items-center gap-3">
+          <ButtonLink id="cancel-button" to="/personal-information">
```

Comment:
Since we have the previous page (ie. `/demographics-part1`), we should link to that page instead. We would need to retrieve the `id` and construct the URL with this `id` (ie. `/intake/{flowId}/demographics-part1`). 

Take a look at [`/_public+/intake+/$id+/email.tsx`](https://github.com/DTS-STN/canadian-dental-care-plan/blob/dd8d179edeea38ea36d9b7f256d05cdd6d5b31ed/frontend/app/routes/_public%2B/intake%2B/%24id%2B/email.tsx#L23) to see how we can do this.

Note that we may change this in the future so that "Back" button saves the form without validation, but we can worry about that in a future PR if needed.

Reply:
please take a look at the code i came up with